src/Command/SingleDirectoryComponent.php

```diff
@@ -0,0 +1,218 @@
+<?php declare(strict_types = 1);
+
+namespace DrupalCodeGenerator\Command;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Extension\ThemeHandlerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Asset\LibraryDiscovery;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Extension\ExtensionLifecycle;
+use DrupalCodeGenerator\Application;
+use DrupalCodeGenerator\Asset\AssetCollection;
+use DrupalCodeGenerator\Asset\File;
+use DrupalCodeGenerator\Attribute\Generator;
+use DrupalCodeGenerator\GeneratorType;
+use DrupalCodeGenerator\InputOutput\Interviewer;
+use DrupalCodeGenerator\Utils;
+use DrupalCodeGenerator\Validator\Choice;
+use DrupalCodeGenerator\Validator\Optional;
+use DrupalCodeGenerator\Validator\Required;
+use DrupalCodeGenerator\Validator\RequiredMachineName;
+use Symfony\Component\Console\Question\Question;
+
+#[Generator(
+  name: 'single-directory-component',
+  description: 'Generates Drupal SDC theme component',
+  aliases: ['sdc'],
+  templatePath: Application::TEMPLATE_PATH . '/_sdc',
+  type: GeneratorType::THEME,
```

Comment:
This type stands for new themes. 
SDC generator needs to be defined as `GeneratorType::THEME_COMPONENT`.